iree-jax/benchmark/benchmark_model.py

```diff
@@ -52,18 +56,41 @@ def dump_result(file_path: str, result: dict) -> None:
 def bytes_to_mb_str(bytes: Optional[int]) -> str:
   return "n/a" if bytes is None else f"{bytes / 1e6:.6f}"
 
-
-def run_framework_benchmark(model_name: str, model_class: Any, batch_size: int,
+def retrieve_data(model_data: data_types.ModelData, cache_dir: str) -> tuple[Any, ...]:
+  data = ()
+  for url in model_data.source_url:
+    assert url.startswith("https://storage.googleapis.com/iree-model-artifacts/")
+    relative_path = url.removeprefix("https://storage.googleapis.com/iree-model-artifacts/")
+    local_path = cache_dir / relative_path
+
+    if not os.path.exists(local_path):
+      pathlib.Path(os.path.dirname(local_path)).mkdir(parents=True, exist_ok=True)
+      r = requests.get(url)
+      print(f"Downloading {url}")
+      open(local_path, 'wb').write(r.content)
```

Comment:
This might also buffer all data in memory. It's fine if the input/output data size is not too large.

Reply:
Changed it to stream and save chunks.